### ISSUE TITLE
userId in records not correct

### ISSUE THREAD

**Issue by sboettcher** [RADAR-CNS/RADAR-Commons-Android#11]
I noticed an issue with the record keys' userId field, where the string that is sent is the default string entry from `remote_config_defaults.xml` regardless of what is set via the GUI 'Patient ID' field. <s>This happens only after the UDOO is rebooted and the app starts automatically. Before (e.g. after installing it via Android Studio) it works as expected. It still happens when I start it again by any means.</s>

Commons-Android Rev: `6056547`
Commons Rev: `791ed20`
AndroidApplication Rev: `1862a66`

**Comment by sboettcher** [comment]
Looks like I was wrong, it also happens when restarting the app via any means, not just rebooting. The only solution atm is to set the ID again via the GUI.
I have traced the issue to the `connection.hasService()` check in [DetailMainActivityView.java](https://github.com/RADAR-CNS/RADAR-AndroidApplication/blob/dev/app/src/main/java/org/radarcns/DetailMainActivityView.java#L263) in the AndroidApplication repo. Maybe this issue should be moved there, sorry about that.
When the check occurs, the services have not yet been bound, see also this log output:

```
05-31 12:02:08.300 22280-22280/org.radarcns.android.android I/DetailMainActivityView: DEBUG checking service for DeviceServiceConnection<org.radarcns.application.ApplicationStatusService>
05-31 12:02:08.301 22280-22280/org.radarcns.android.android I/DetailMainActivityView: DEBUG checking service for DeviceServiceConnection<org.radarcns.phone.PhoneSensorService>
05-31 12:02:08.301 22280-22280/org.radarcns.android.android I/DetailMainActivityView: DEBUG checking service for DeviceServiceConnection<org.radarcns.empatica.E4Service>
05-31 12:02:08.301 22280-22280/org.radarcns.android.android I/DetailMainActivityView: DEBUG checking service for DeviceServiceConnection<org.radarcns.biovotion.BiovotionService>
05-31 12:02:08.301 22280-22280/org.radarcns.android.android I/DetailMainActivityView: DEBUG checking service for DeviceServiceConnection<org.radarcns.eegsync.EEGSyncService>
05-31 12:02:10.136 22280-22280/org.radarcns.android.android I/BaseServiceConnection: Bound to service ComponentInfo{org.radarcns.android.android/org.radarcns.application.ApplicationStatusService}
05-31 12:02:10.328 22280-22280/org.radarcns.android.android I/BaseServiceConnection: Bound to service ComponentInfo{org.radarcns.android.android/org.radarcns.phone.PhoneSensorService}
05-31 12:02:10.377 22280-22280/org.radarcns.android.android I/BaseServiceConnection: Bound to service ComponentInfo{org.radarcns.android.android/org.radarcns.empatica.E4Service}
05-31 12:02:10.405 22280-22280/org.radarcns.android.android I/BaseServiceConnection: Bound to service ComponentInfo{org.radarcns.android.android/org.radarcns.biovotion.BiovotionService}
05-31 12:02:10.626 22280-22280/org.radarcns.android.android I/BaseServiceConnection: Bound to service ComponentInfo{org.radarcns.android.android/org.radarcns.eegsync.EEGSyncService}
--- setting ID via GUI ---
05-31 12:02:25.163 22280-22280/org.radarcns.android.android I/DetailMainActivityView: DEBUG checking service for DeviceServiceConnection<org.radarcns.application.ApplicationStatusService>
05-31 12:02:25.164 22280-22280/org.radarcns.android.android I/DetailMainActivityView: DEBUG setting userId
05-31 12:02:25.165 22280-22280/org.radarcns.android.android I/DetailMainActivityView: DEBUG checking service for DeviceServiceConnection<org.radarcns.phone.PhoneSensorService>
05-31 12:02:25.165 22280-22280/org.radarcns.android.android I/DetailMainActivityView: DEBUG setting userId
05-31 12:02:25.166 22280-22280/org.radarcns.android.android I/DetailMainActivityView: DEBUG checking service for DeviceServiceConnection<org.radarcns.empatica.E4Service>
05-31 12:02:25.166 22280-22280/org.radarcns.android.android I/DetailMainActivityView: DEBUG setting userId
05-31 12:02:25.166 22280-22280/org.radarcns.android.android I/DetailMainActivityView: DEBUG checking service for DeviceServiceConnection<org.radarcns.biovotion.BiovotionService>
05-31 12:02:25.166 22280-22280/org.radarcns.android.android I/DetailMainActivityView: DEBUG setting userId
05-31 12:02:25.167 22280-22280/org.radarcns.android.android I/DetailMainActivityView: DEBUG checking service for DeviceServiceConnection<org.radarcns.eegsync.EEGSyncService>
05-31 12:02:25.167 22280-22280/org.radarcns.android.android I/DetailMainActivityView: DEBUG setting userId
```

**Comment by blootsvoets** [comment]
Thanks for the bug report, is this solved by adding in `DetailMainActivityView.java` the line:
```java
radarConfiguration.put(DEFAULT_GROUP_ID_KEY, userId);
```

between the lines:

```java
preferences.edit().putString("userId", userId).apply();
mGroupIdInput.setText(userId);
```

**Comment by sboettcher** [comment]
Getting
```
java.lang.IllegalStateException: Cannot start recording: user ID is not set.
 	at org.radarcns.android.device.DeviceService.startRecording(DeviceService.java:381)
...
```
with that

**Comment by sboettcher** [comment]
Ah but putting
```
radarConfiguration.put(RADAR_PREFIX + DEFAULT_GROUP_ID_KEY, userId);
```
works :)

**Comment by sboettcher** [comment]
Was a bit too quick with closing, the original issue still persists :(

**Comment by blootsvoets** [comment]
It should be without the prefix. Additional fix: in `DetailMainActivity.java` replace the current createConfiguration method with the following code:
```java
    @Override
    protected RadarConfiguration createConfiguration() {
        // TODO: turn off developer mode
        RadarConfiguration configuration = RadarConfiguration.configure(this, true, R.xml.remote_config_defaults);
        SharedPreferences preferences = getSharedPreferences("main", Context.MODE_PRIVATE);
        String userId = preferences.getString("userId", "");
        if (!userId.isEmpty()) {
            configuration.put(RadarConfiguration.DEFAULT_GROUP_ID_KEY, userId);
        }
        return configuration;
    }
```

**Comment by sboettcher** [comment]
Still the same behaviour. Maybe @fnobilia could try to reproduce this with their setup, to make sure it is not my setup...
Also still have to add the prefix to not get the `IllegalStateException`. I will check again to see if I have the latest relevant revisions.

I have `ID1` entered in the config xml, and `ID2` entered via the GUI. When I start the app the userId read from the SharedPreferences object is `ID2` as expected, but ```RadarConfiguration.getStringExtra(bundle, DEFAULT_GROUP_ID_KEY)```
in [DeviceService.java](https://github.com/RADAR-CNS/RADAR-Commons-Android/blob/dev/src/main/java/org/radarcns/android/device/DeviceService.java#L637) returns `ID1`

**Comment by blootsvoets** [comment]
Looks like the configuration does not get completely passed to the service using the bundle, only using the direct userId call. Fixed in #12.

**Comment by sboettcher** [comment]
can confirm that https://github.com/RADAR-CNS/RADAR-Commons-Android/pull/12 in combination with https://github.com/RADAR-CNS/RADAR-Commons-Android/issues/11#issuecomment-305151547 fixes.
